imap_processing/tests/codice/test_codice_l1a.py

```diff
@@ -10,139 +9,172 @@
-    "imap_codice_l1a_hi-counters-aggregated",
-    "imap_codice_l1a_hi-counters-singles",
-    "imap_codice_l1a_hi-omni",
-    "imap_codice_l1a_hi-sectored",
-    "imap_codice_l1a_hi-pha",
     "imap_codice_l1a_lo-counters-aggregated",
     "imap_codice_l1a_lo-counters-singles",
-    "imap_codice_l1a_lo-sw-angular",
-    "imap_codice_l1a_lo-nsw-angular",
     "imap_codice_l1a_lo-sw-priority",
     "imap_codice_l1a_lo-nsw-priority",
     "imap_codice_l1a_lo-sw-species",
     "imap_codice_l1a_lo-nsw-species",
+    "imap_codice_l1a_lo-sw-angular",
+    "imap_codice_l1a_lo-nsw-angular",
+    "imap_codice_l1a_hi-counters-aggregated",
+    "imap_codice_l1a_hi-counters-singles",
+    "imap_codice_l1a_hi-omni",
+    "imap_codice_l1a_hi-sectored",
+    "imap_codice_l1a_hi-priority",
     "imap_codice_l1a_lo-pha",
+    "imap_codice_l1a_hi-pha",
 ]
+
 EXPECTED_NUM_VARIABLES = [
-    129,  # hskp
-    1,  # hi-counters-aggregated
-    3,  # hi-counters-singles
-    8,  # hi-omni
-    4,  # hi-sectored
-    0,  # hi-pha
+    0,  # hi-ialirt  # TODO: Need to implement
+    0,  # lo-ialirt  # TODO: Need to implement
+    148,  # hskp
     3,  # lo-counters-aggregated
     3,  # lo-counters-singles
-    6,  # lo-sw-angular
-    3,  # lo-nsw-angular
     7,  # lo-sw-priority
     4,  # lo-nsw-priority
     18,  # lo-sw-species
     10,  # lo-nsw-species
-    0,  # lo-pha
+    6,  # lo-sw-angular
+    3,  # lo-nsw-angular
+    1,  # hi-counters-aggregated
+    3,  # hi-counters-singles
+    8,  # hi-omni
+    4,  # hi-sectored
+    0,  # hi-priority  # TODO: Need to implement
+    0,  # lo-pha  # TODO: Need to implement
+    0,  # hi-pha  # TODO: Need to implement
 ]
 
 
-@pytest.fixture(params=TEST_PACKETS)
-def test_l1a_data(request) -> xr.Dataset:
+@pytest.fixture(scope="session")
+def test_l1a_data() -> xr.Dataset:
     """Return a ``xarray`` dataset containing test data.
 
     Returns
     -------
-    dataset : xarray.Dataset
-        A ``xarray`` dataset containing the test data
+    processed_datasets : list[xarray.Dataset]
+        A list of ``xarray`` datasets containing the test data
     """
 
-    dataset = process_codice_l1a(file_path=request.param, data_version="001")
+    processed_datasets = process_codice_l1a(file_path=TEST_L0_FILE, data_version="001")
+
+    return processed_datasets
+
+
+@pytest.mark.parametrize("index", range(len(EXPECTED_ARRAY_SHAPES)))
+def test_l1a_data_array_shape(test_l1a_data, index):
+    """Tests that the data arrays in the generated CDFs have the expected shape.
+
+    Parameters
+    ----------
+    test_l1a_data : list[xarray.Dataset]
+        A list of ``xarray`` datasets containing the test data
+    index : int
+        The index of the list to test
+    """
+
+    processed_dataset = test_l1a_data[index]
+    expected_shape = EXPECTED_ARRAY_SHAPES[index]
+
+    # Mark currently broken/unsupported datasets as expected to fail
+    # TODO: Remove these once they are supported
+    if index in [0, 1, 15]:
```

Comment:
Yes, good catch. The test passes for those because I set the expected number of variables to 0 for now, but I should add the indices to this list for consistency.